executor/insert_common.go

```diff
@@ -60,6 +60,15 @@ type InsertValues struct {
 	colDefaultVals  []defaultVal
 	evalBuffer      chunk.MutRow
 	evalBufferTypes []*types.FieldType
+
+	// cacheHasValue cache haveValue for consecutive autoid batch alloc.
```

Comment:
No need cache, just calculate in `lazyAdjustAutoIncrementDatum `?

Reply:
Nice code comprehension! Addressed!